packages/flutter_tools/test/ios/devices_test.dart

```diff
@@ -183,4 +185,42 @@ f577a7903cc54959be2e34bc4f7f80b7009efcf4
       IMobileDevice: () => mockIMobileDevice,
     });
   });
+
+  testUsingContext('IOSDevice.isSupportedForProject is true on module project', () async {
+    fs.file('pubspec.yaml')
+      ..createSync()
+      ..writeAsStringSync(r'''
+name: example
+
+flutter:
+  module: {}
+''');
+    fs.file('.packages').createSync();
+    final FlutterProject flutterProject = await FlutterProject.current();
+
+    expect(IOSDevice('test').isSupportedForProject(flutterProject), true);
```

Comment:
same question.

Reply:
We support a "module" style application where flutter is hosted as part of an existing native application instead of containing an editable sub-app directory. This includes hidden .ios and .android folders which contain fully generated code necessary to link the dart + native projects.

The idea was that these folders would be fully managed and regenerated by the framework, so for the sake of backwards compatibility we cannot use the presence of the folders as a hint. Instead, the pubspec will contain a `module: ...more_info` field